vm/src/vm.rs

```diff
@@ -460,11 +478,16 @@ impl VirtualMachine {
 
     fn new_instance(&mut self, type_ref: PyObjectRef, args: PyFuncArgs) -> PyResult {
         // more or less __new__ operator
-        let obj = PyObject::new(PyObjectKind::None, type_ref.clone());
+        let dict = self.new_dict();
+        let obj = PyObject::new(PyObjectKind::Instance{dict: dict}, type_ref.clone());
+        let init = type_ref.get_attr(&String::from("__init__"));
+        let mut self_args = PyFuncArgs { args: args.args };
+        self_args.args.insert(0, obj.clone());
+        self.invoke(init, self_args).unwrap();
```

Comment:
Here, the result must be checked, and if an exception is there, it must be returned.

Reply:
I'll push a fix to master.